### PR TITLE
feat: add Kimi Linear model support

### PR DESCRIPTION
**Pull request by HuiyingLi**
## Summary

Adds native Automodel support for `moonshotai/Kimi-Linear-48B-A3B-Instruct`:

- Implements the Kimi Linear config, causal LM, KDA attention, MLA attention, HF state-dict adapter, and registry wiring.
- Uses FLA KDA kernels (`fused_kda_gate`, `chunk_kda` / `fused_recurrent_kda`, `ShortConvolution`, `FusedRMSNormGated`) following the HF `trust_remote_code` implementation.
- Keeps KDA `A_log` and `dt_bias` in fp32 via an FSDP-safe holder.
- Adds Kimi-specific MoE routing switches needed for HF parity and EP training, while preserving default MoE behavior.
- Adds an EP=8, PP=1 HellaSwag fine-tuning recipe for the 48B-A3B checkpoint.

## HF Parity

First 8 layers matched against Hugging Face `trust_remote_code=True` implementation for `moonshotai/Kimi-Linear-48B-A3B-Instruct`.

- `kl_batchmean`: `-1.949409e-06` (numerical noise around zero)
- `kl_per_token`: `-1.522976e-08`
- `max_abs_diff`: `0.0`
- `mean_abs_diff`: `0.0`
- logits shape: `[1, 128, 163840]`

The tensor bisect ended with layers 0 through 7 exact on logits and routed MoE gate indices/weights.

## HellaSwag 100-step W&B Run

W&B: https://wandb.ai/Nemo-automodel/automodel-kimi-linear/runs/l4l4eb7d

image

Run setup: single node, 8x H100 80GB, FSDP2, EP=8, PP=1, `kda_mode=chunk`, `kda_unpad_inputs=false`, W&B online.

- step 0 train loss: `2.7050`
- validation loss trend: `3.2480` at step 9 -> `3.0102` at step 99
- final step 99 train loss: `1.8430`
- final W&B summary: `_step=99`, `loss=1.8429668`, `val_loss=3.0102007`, `grad_norm=5.1753`

## Validation

- `python -m py_compile ...` on touched model/MoE/test files
- `python -m ruff check ...` on touched Python files
- `python -m ruff format --check ...` on touched Python files
- `python tools/lint_example_yamls.py --automodel-dir . examples/llm_finetune/kimi/kimi_linear_48b_a3b_hellaswag.yaml`
- `CUDA_VISIBLE_DEVICES= PYTHONPATH=$PWD python -m pytest -q tests/unit_tests/models/kimi_linear tests/unit_tests/_transformers/test_registry.py` (`34 passed`)
- `CUDA_VISIBLE_DEVICES= PYTHONPATH=$PWD python -m pytest -q tests/unit_tests/moe/test_experts.py::TestGroupedExpertsRouteWeightAfterDown tests/unit_tests/moe/test_layers.py::TestGate::test_gate_router_weights_fp32_keeps_precision_output tests/unit_tests/moe/test_layers.py::TestGate::test_gate_router_weights_can_use_score_correction_bias` (`5 passed`)

### CODE REVIEW

**Comment by HuiyingLi** [comment]
/claude review\n\nLatest head: 98cb18c8d3089b5e187b072f4c6c74fd57f72f3b

**Comment by HuiyingLi** [comment]
/claude review

Latest head: 98cb18c8d3089b5e187b072f4c6c74fd57f72f3b

**Comment by HuiyingLi** [comment]
closing as already included in #3243
